### ISSUE TITLE
Twig Sync Loading - In Browser?

### ISSUE THREAD

**Issue by cwbeck** [twigjs/twig.js#426]
Extending a template works in Chrome, but does produce the following error:

<img width="693" alt="screen shot 2016-11-02 at 11 18 57" src="https://cloud.githubusercontent.com/assets/425262/19926807/9e1fa124-a0ee-11e6-8221-8c90d7c9397a.png">

This should be all async, so why is this blocking? Anyway to fix this?

```
var template = twig.twig({
    id: "login",
    namespaces: { 'views_dir': 'js/app/' },
    href: "js/app/pages/login/login.twig",
    async: true,
    load: function(template) {
        console.log(template.render());
    }
});
```

Where js/app/pages/login/login.twig -> extends another template

**Comment by DrParanoia** [comment]
I can confirm this.

The first template is loaded asynchronously, however if you have any includes that requires additional templates to be loaded, these templates are loaded in a synchronous manner. To make matters worse it freezes the UI until templates are loaded.

**Comment by dave-irvine** [comment]
Can you give me an example of the two templates? Just minimum breaking example.

**Comment by cwbeck** [comment]
Any template extending another template. Where first template, in my case ```login.twig``` can just contain:

```
{% extends "base.twig" %}
...
```

And `base.twig` contains:

```
<b>hello world!</b>
```

It forces a sync request for subsequent templates rather than a fetch as an async waterfall which is the desired pattern and thus not blocking the page. templates must be fetched in order as dependencies are unknown until point of render on each.

Hope that helps.

**Comment by cwbeck** [comment]
Is this still an active project?

**Comment by dave-irvine** [comment]
As much as it can be with one maintainer, yes. Happy for a pull request.

**Comment by DrParanoia** [comment]
When async loading the template, all included or extended templates that it has will not load in async mode.

I am afraid that fixing this is not a trivial task :/ But I just have to ask - is there any hope that this will be fixed in the near future?

**Comment by dave-irvine** [comment]
I'd love to fix it, but I'm not currently using Twig in any of my projects at my job, or any of my spare-time projects. As such I don't have much time to devote to twig.js other than attending to pull requests and cutting releases.

If someone else can fix this I'm more than happy to help get it landed, but I don't have any time to dedicate to it at the moment.

I think @olets uses Twig.js for some of his work projects, he might be a candidate for fixing :)

**Comment by cwbeck** [comment]
@DrParanoia I've spent a bit of time on this, you are correct, not a simple fix. Requires quite a bit of logic change.

@dave-irvine - spent a bit of time on this, but I'm just not familiar enough with the internals of the project. Perhaps i've approached some bits wrong.

Would love to submit a PR, but I've not got enough time to fully get into this.

**Comment by DrParanoia** [comment]
@cwbeck Yeah, I was trying to fix it myself, but it's a bit complicated and requires quite a bit of studying the code. Maybe I will try to do something about it in the future... Right now, the best thing to do (I think) is to pre-load all included templates in reverse order on page load, so that all included templates are already in the cache when they are required, otherwise the browser will lock.

**Comment by willrowe** [comment]
Would it be possible for someone to submit a PR with a failing test for this?
